[Bug] Golang CI failed

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)


### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.


### Describe the bug

```
# golang.org/x/sys/unix
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/syscall.go:83:16: undefined: unsafe.Slice
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/syscall_linux.go:2270:9: undefined: unsafe.Slice
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/syscall_unix.go:118:7: undefined: unsafe.Slice
Error: /home/runner/go/pkg/mod/golang.org/x/sys@v0.4.0/unix/sysvshm_unix.go:33:7: undefined: unsafe.Slice

```
Detailed info could be found in https://github.com/apache/incubator-uniffle/actions/runs/4031027397/jobs/6930075459

### Affects Version(s)

master

### Uniffle Server Log Output

_No response_

### Uniffle Engine Log Output

_No response_

### Uniffle Server Configurations

_No response_

### Uniffle Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

--- a/.github/workflows/parallel.yml
+++ b/.github/workflows/parallel.yml
@@ -78,10 +78,11 @@ jobs:
         java-version: ${{ inputs.java-version }}
         distribution: ${{ inputs.jdk-distro }}
         cache: maven
-    - name: Set up Go 1.16
+    - name: Set up Go 1.17
+      if: ${{ matrix.profile == "kubernetes" }}
       uses: actions/setup-go@v3
       with:
-        go-version: 1.16
+        go-version: 1.17
     - name: Execute maven with -P${{ matrix.profile }}
       if: ${{ !inputs.experimental }}
       run: mvn -B -fae -P${{ matrix.profile }} ${{ inputs.maven-args }} | tee /tmp/maven.log

--- a/.github/workflows/parallel.yml
+++ b/.github/workflows/parallel.yml
@@ -79,7 +79,7 @@ jobs:
         distribution: ${{ inputs.jdk-distro }}
         cache: maven
     - name: Set up Go 1.17
-      if: ${{ matrix.profile == "kubernetes" }}
+      if: ${{ matrix.profile == 'kubernetes' }}
       uses: actions/setup-go@v3
       with:
         go-version: 1.17

